File: src/main/java/com/jif/estoque/dto/ProdutoDTO.java
package com.jif.estoque.dto;

import com.jif.estoque.enums.Unidade;
import lombok.Data;

import java.math.BigDecimal;

@Data
public class ProdutoDTO {

    private Long id;
    private String codigo;
    private String nome;
    private String descricao;
    private String categoria;
    private Unidade unidade;
    private BigDecimal precoCusto;
    private BigDecimal estoqueMin;
    private Long fornecedorId;
    private Boolean ativo;
}

File: frontend/src/app/funcionalidades/produtos/store/Produto.dto.ts
import { Unidade } from '../../../enums/enums';

export interface ProdutoDTO {
  id?: number;
  codigo: string;
  nome: string;
  descricao: string;
  categoria: string;
  unidade: Unidade;
  precoCusto: number;
  estoqueMin: number;
  fornecedorId: number;
  ativo: boolean;
}

export interface ProdutoDetalheDTO {
  id: number;
  codigo: string;
  nome: string;
  descricao: string;
  categoria: string;
  unidade: Unidade;
  precoCusto: number;
  estoqueMin: number;
  ativo: boolean;
  nomeFornecedor: string;
  quantidade: number;
  custoMedio: number;
}

export interface ProdutoSaldoDTO {
  id: number;
  codigo: string;
  nome: string;
  estoqueMin: number;
  quantidadeAtual: number;
}
